### ISSUE TITLE
DataType.convert(Object...) and convert(Collection<?>) don't work for converted data types

### ISSUE THREAD

**Issue by AnghelLeonard** [jOOQ/jOOQ#11168]
### Expected behavior
The `DataType.convert​(Object... objects)` should convert the array of `objects` and `DataType.convert​(Collection<?> objects)` should convert the collection of `objects`. In my particular case, `objects` is `Integer[]` respectively, `List<Integer>` and the expected result is `YearMonth[]`, respectively `List<YearMonth>`, (`java.time.YearMonth`).
### Actual behavior
Throw `DataTypeException`: Cannot convert from 201212 (class `java.lang.Integer`) to class `java.time.YearMonth`

### Steps to reproduce the problem

I have this converter:

```Java
import java.time.YearMonth;
import org.jooq.Converter;
import org.jooq.DataType;
import static org.jooq.impl.SQLDataType.INTEGER;

public class YearMonthConverter implements Converter<Integer, YearMonth> {

    public static final Converter<Integer, YearMonth> INTEGER_YEARMONTH_CONVERTER
            = new YearMonthConverter();

    public static final Converter<Integer[], YearMonth[]> INTEGER_YEARMONTH_ARR_CONVERTER
            = INTEGER_YEARMONTH_CONVERTER.forArrays();

    public static final DataType<YearMonth> YEARMONTH
            = INTEGER.asConvertedDataType(INTEGER_YEARMONTH_CONVERTER);
    
    @Override
    public YearMonth from(Integer t) {

        if (t != null) {
            return YearMonth.of(t / 100, t % 100);
        }

        return null;
    }

    @Override
    public Integer to(YearMonth u) {

        if (u != null) {

            return (u.getYear() * 100)
                    + u.getMonth().getValue();
        }

        return null;
    }

    @Override
    public Class<Integer> fromType() {
        return Integer.class;
    }

    @Override
    public Class<YearMonth> toType() {
        return YearMonth.class;
    }

}
```

Preparation:

```Java
Integer[] arr = {201212, 202011}; // Given: 201212, Expected: 2012-12
List<Integer> list = List.of(arr);

// these are listed above in the converter as well
public static final Converter<Integer, YearMonth> INTEGER_YEARMONTH_CONVERTER
            = new YearMonthConverter();

public static final DataType<YearMonth> YEARMONTH
            = INTEGER.asConvertedDataType(INTEGER_YEARMONTH_CONVERTER);
```

And, here it is the results of using the converter via `YEARMONTH`:

```Java
YearMonth ym = YEARMONTH.convert(arr[0]);          // Work, result: 2012-12
YearMonth[] yma = YEARMONTH.convert(arr);          // DataTypeException
List<YearMonth> yml = YEARMONTH.convert(list);     // DataTypeException
```
Note: Using `INTEGER_YEARMONTH_ARR_CONVERTER` works for arrays.

### Versions

- jOOQ: 3.14.4
- Java: 14
- Database (include vendor): MySQL
- OS: Windows 10
- JDBC Driver (include name if inofficial driver): latest JDBC driver for MySQL

**Comment by lukaseder** [comment]
Thanks for your report. I can reproduce this problem

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11170). Thanks again for your report!
